urnai/actions/chain_of_actions_base.py

```diff
@@ -0,0 +1,26 @@
+from abc import ABC
+
+from urnai.actions.action_base import ActionBase
+
+
+class ChainOfActionsBase(ABC):
+
+	"""This class works as a sequence of actions"""
+	
+	action_list: list[ActionBase]
+	
+	def __init__(self):
+		self.action_list = []
+	
+	def get_action(self, action_index) -> ActionBase:
+		if action_index < self.length:
+			self.action_list[action_index]
+	
+	def check(self) -> bool:
+		""" Checking all of the actions in the sequence """
+		
+		return all(action.check() for action in self.action_list)
```

Comment:
Something that crossed my mind about this method is that there may be cases where an action can't be executed in that iteration, but would be available in the next one. Using `all` would then return `False`, because all `check()` must be `True` for it to return `True`.

It would be interesting to discuss this function at the next meeting to see if this logic is appropriate or not.